clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -2216,3 +2305,18 @@ mlir::Value ScalarExprEmitter::VisitUnaryExprOrTypeTraitExpr(
   return Builder.getConstInt(CGF.getLoc(E->getSourceRange()),
                              E->EvaluateKnownConstInt(CGF.getContext()));
 }
+
+mlir::Value CIRGenFunction::buildCheckedInBoundsGEP(
+    mlir::Type ElemTy, mlir::Value Ptr, ArrayRef<mlir::Value> IdxList,
+    bool SignedIndices, bool IsSubtraction, SourceLocation Loc) {
+  mlir::Type PtrTy = Ptr.getType();
+  assert(IdxList.size() == 1 && "multi-index ptr arithmetic NYI");
+  mlir::Value GEPVal = builder.create<mlir::cir::PtrStrideOp>(
+      CGM.getLoc(Loc), PtrTy, Ptr, IdxList[0]);
+
+  // If the pointer overflow sanitizer isn't enabled, do nothing.
+  if (!SanOpts.has(SanitizerKind::PointerOverflow))
+    return GEPVal;
+
+  llvm_unreachable("pointer overflow sanitizer NYI");
```

Comment:
The unreachable is good but you should also add an unimplemented feature here!

Reply:
Added!